token-metadata/program/src/processor/mod.rs

```diff
@@ -435,3 +450,24 @@ fn has_programmable_metadata<'a>(
 
     Ok(false)
 }
+
+/// Checks if the instruction's accounts contain a pNFT metadata.
+///
+/// We need to determine if we are dealing with a pNFT metadata or not
+/// so we can restrict the available instructions.
+fn is_locked<'a>(program_id: &Pubkey, accounts: &'a [AccountInfo]) -> Result<bool, ProgramError> {
```

Comment:
This does not need to return a `Result` as there's no possible error condition. Just return a `bool` and that will also make the logic in the processor cleaner.

Reply:
Good point, returning a `bool` now.